ui/app/(core)/networking/page.tsx

```diff
@@ -0,0 +1,617 @@
+
+// NAT (global)
+import { getNATInfo, updateNATInfo } from "@/queries/nat";
+
+// Shared UI
+import DeleteConfirmationModal from "@/components/DeleteConfirmationModal";
+import EmptyState from "@/components/EmptyState";
+
+import type { DataNetwork, Route } from "@/types/types";
+
+const MAX_WIDTH = 1400;
+
+type TabKey = "data-networks" | "routes" | "nat";
+
+export default function NetworkingPage() {
+  const theme = useTheme();
+  const { role, accessToken } = useAuth();
+  const canEdit = role === "Admin" || role === "Network Manager";
+
+  const [tab, setTab] = useState<TabKey>("data-networks");
+
+  // ------------ Alerts (section-scoped) ------------
+  const [dnAlert, setDnAlert] = useState<{
+    message: string;
+    severity: "success" | "error" | null;
+  }>({ message: "", severity: null });
+  const [rtAlert, setRtAlert] = useState<{
+    message: string;
+    severity: "success" | "error" | null;
+  }>({ message: "", severity: null });
+  const [natAlert, setNatAlert] = useState<{
+    message: string;
+    severity: "success" | "error" | null;
+  }>({ message: "", severity: null });
+
+  // ------------ Data Networks ------------
+  const {
+    data: dataNetworks = [],
+    isLoading: dnLoading,
+    refetch: refetchDataNetworks,
+  } = useQuery<DataNetwork[]>({
+    queryKey: ["data-networks", accessToken],
+    queryFn: () => listDataNetworks(accessToken || ""),
+    refetchInterval: 5000,
+    refetchIntervalInBackground: true,
+    refetchOnWindowFocus: true,
+  });
+
+  const [isCreateDNOpen, setCreateDNOpen] = useState(false);
+  const [isEditDNOpen, setEditDNOpen] = useState(false);
+  const [isDeleteDNOpen, setDeleteDNOpen] = useState(false);
+  const [editDN, setEditDN] = useState<DataNetwork | null>(null);
+  const [selectedDNName, setSelectedDNName] = useState<string | null>(null);
+
+  const handleOpenCreateDN = () => setCreateDNOpen(true);
+  const handleCloseCreateDN = () => setCreateDNOpen(false);
+  const handleEditDN = (dn: DataNetwork) => {
+    setEditDN(dn);
+    setEditDNOpen(true);
+  };
+  const handleRequestDeleteDN = (name: string) => {
+    setSelectedDNName(name);
+    setDeleteDNOpen(true);
+  };
+  const handleConfirmDeleteDN = async () => {
+    setDeleteDNOpen(false);
+    if (!selectedDNName || !accessToken) return;
+    try {
+      await deleteDataNetwork(accessToken, selectedDNName);
+      setDnAlert({
+        message: `Data Network "${selectedDNName}" deleted successfully!`,
+        severity: "success",
+      });
+      refetchDataNetworks();
+    } catch (error: unknown) {
+      setDnAlert({
+        message: `Failed to delete data network "${selectedDNName}": ${String(error)}`,
+        severity: "error",
+      });
+    } finally {
+      setSelectedDNName(null);
+    }
+  };
+
+  const dnDescription = useMemo(
+    () =>
+      "Manage the IP networks used by your subscribers. Data Network Names (DNNs) define connectivity per slice/DNN.",
```

Comment:
The second sentence seems a bit recursive. I am not quite sure what you wish to express here. Maybe something like "Data Network Names (DNNs) define network configuration for a group of subscribers"?